### PR TITLE
React tabs with router

### PR DESCRIPTION
**Pull request by vsupruniuk**
[DEMO LINK](https://vsupruniuk.github.io/react_tabs-with-router/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,31 +1,21 @@
...
+
+export const App: React.FC = () => {
+  return (
+    <div className="App">
+      <Navigation />
+
+      <Routes>
+        <Route path="/" element={<HomePage />} />
+        <Route path="/tabs" element={<TabsPage />} />
+        <Route path="/tabs/:tabId" element={<TabsPage />} />
```
Comment (jstmpelowycz) [inline src/App.tsx:16]: you'd rather use nested routes

File: src/Components/Navigation/Navigation.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  return (
+    <ul className="nav nav-pills">
+      <li className="nav-item">
+        <NavLink
+          className={(navData) => (classNames(
+            'nav-link',
+            {
+              active: navData.isActive,
+            },
+          ))}
```
Comment (jstmpelowycz) [inline src/Components/Navigation/Navigation.tsx:15]: looks like a separate callback

File: src/Components/TabsPage/TabsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
+  { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
+  { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
+];
+
+export const TabsPage: React.FC = React.memo(() => {
+  const match = useMatch('/tabs/:tabId');
+  const selectedTabId = match?.params.tabId;
+
+  const getContent = (tabId: string | undefined) => {
```
Comment (jstmpelowycz) [inline src/Components/TabsPage/TabsPage.tsx:15]:
```suggestion
  const getContent = (tabId?: string) => {
```

File: src/Components/TabsPage/TabsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import classNames from 'classnames';
+
+const tabs = [
+  { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
+  { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
+  { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
+];
+
+export const TabsPage: React.FC = React.memo(() => {
+  const match = useMatch('/tabs/:tabId');
```
Comment (jstmpelowycz) [inline src/Components/TabsPage/TabsPage.tsx:12]: you'd rather apply `useParams`

**Review by jstmpelowycz — CHANGES_REQUESTED** [review]

**Review by jstmpelowycz — CHANGES_REQUESTED** [review]

File: src/Components/Navigation/Navigation.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+
+export const Navigation: React.FC = React.memo(() => {
+  const createClassNames = ({ isActive }: { isActive: boolean }) => {
+    if (isActive) {
+      return 'nav-link active';
+    }
+
+    return 'nav-link';
```
[the remark this answers is not in the document]
Comment (vsupruniuk) [reply src/Components/Navigation/Navigation.tsx:10]: 👀

File: src/Components/Navigation/Navigation.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+
+export const Navigation: React.FC = React.memo(() => {
+  const createClassNames = ({ isActive }: { isActive: boolean }) => {
```
Comment (jstmpelowycz) [inline src/Components/Navigation/Navigation.tsx:5]: :harold:

**Review by jstmpelowycz — APPROVED** [review]

**Pull request closed without merging**